server/api/v1/instance_service.go

```diff
@@ -0,0 +1,125 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/pkg/errors"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/plugin/db"
+	v1pb "github.com/bytebase/bytebase/proto/generated-go/v1"
+	"github.com/bytebase/bytebase/store"
+)
+
+const instanceNamePrefix = "instances/"
+
+// InstanceService implements the instance service.
+type InstanceService struct {
+	v1pb.UnimplementedInstanceServiceServer
+	store *store.Store
+}
+
+// NewInstanceService creates a new InstanceService.
+func NewInstanceService(store *store.Store) *InstanceService {
+	return &InstanceService{
+		store: store,
+	}
+}
+
+// GetInstance gets an instance.
+func (s *InstanceService) GetInstance(ctx context.Context, request *v1pb.GetInstanceRequest) (*v1pb.Instance, error) {
+	instanceID, err := getInstanceID(request.Name)
```

Comment:
The name also includes the environments. The instance is a sub-hierarchy of environment.

environments/{environment-id}/instances/{instance-id}